e2e_tests/conftest.py

```diff
@@ -36,19 +131,42 @@ def _create_user_page(username, password) -> Page:
     return _create_user_page
 
 
+@pytest.fixture(scope="session")
+def create_users():
```

Comment:
Do you think the following is simpler?

```
@pytest.fixture(scope="session")
def keycloak_admin():
        realm_name = os.environ.get("KEYCLOAK_REALM_NAME")
        client_id = os.environ.get("KEYCLOAK_CLIENT_ID")
        client_secret = os.environ.get("KEYCLOAK_CLIENT_SECRET")
        server_url = os.environ.get("KEYCLOAK_BASE_URI")

        keycloak_openid = keycloak.KeycloakOpenID(
            server_url=server_url,
            client_id=client_id,
            realm_name=realm_name,
            client_secret_key=client_secret,
        )

        token = keycloak_openid.token(grant_type="client_credentials")
        keycload_admin = keycloak.KeycloakAdmin(
            server_url=server_url,
            realm_name=realm_name,
            token=token,
        )
        return keycload_admin


@pytest.fixture
def create_keycloak_user(keycloak_admin):
    def _create_keycloak_user(groups, user_type):
            user_email = f"{uuid.uuid4().hex}{user_type}@test.com"
            user_pass = uuid.uuid4().hex
            user_first_name = "Test"
            user_last_name = "Name"
            user_id = keycloak_admin.create_user(
                {
                    "firstName": user_first_name,
                    "lastName": user_last_name,
                    "username": user_email,
                    "email": user_email,
                    "enabled": True,
                    "groups": groups,
                    "credentials": [{"value": user_pass, "type": "password"}],
                }
            )
            return user_id, user_email, user_pass
    return _create_keycloak_user

@pytest.fixture(scope="session")
def create_aau_keycloak_user(keycloak_admin):
    user_groups = ["/ayr_user_type/view_all"]
    user_type = "aau"
    user_id, user_email, user_pass = create_keycloak_user(user_groups, user_type)

    yield user_email, user_pass

    keycloak_admin.delete_user(user_id=user_id)

@pytest.fixture(scope="session")
def create_standard_keycloak_user(keycloak_admin):
    user_groups = ["/ayr_user_type/view_dept","/transferring_body_user/Testing A"]
    user_type = "standard"
    user_id, user_email, user_pass = create_keycloak_user(user_groups, user_type)

    yield user_email, user_pass

    keycloak_admin.delete_user(user_id=user_id)

@pytest.fixture
def aau_user_page(create_user_page, create_aau_keycloak_user) -> Page:
    username, password = create_aau_keycloak_user
    page = create_user_page(username, password)
    yield page
    page.goto("/sign-out")


@pytest.fixture
def standard_user_page(create_user_page, create_standard_keycloak_user) -> Page:
    username, password = create_standard_keycloak_user
    page = create_user_page(username, password)
    yield page
    page.goto("/sign-out")
```


But at minimum I think the client should not need to know about the type of user, instead "all_access_user" or just equivalent groups could be passed to `create_user`.

Also there are soe variables unused which I have removed in the rewrite above but if we keep it in the current format we should remove, for example `auth_url`

Reply:
Definitely, will replace it with this 👍

Reply:
@anthonyhashemi pushed these changes with some very minor changes for a couple errors

Reply:
sweet, looks good, approving 👌